Sources/hostmgr-helper/VirtualMachineSlot.swift

```diff
@@ -63,52 +62,53 @@ class VirtualMachineSlot: NSObject, ObservableObject, VZVirtualMachineDelegate {
         } catch {
             Logger.helper.error("Error launching VM: \(error.localizedDescription)")
             Logger.helper.error("Attempting Cleanup of \(launchConfiguration.handle)")
-            try await vmManager.removeVM(name: launchConfiguration.handle)
 
+            try? await VMManager.removeVM(name: launchConfiguration.handle)
             self.status = .crashed(error)
             throw error
         }
     }
 
     @MainActor
     func stopVirtualMachine() async throws {
-        self.status = .stopping
-        try await virtualMachine?.stop()
-        self.status = .empty
+        switch status {
+        case .starting, .running:
+            self.status = .stopping
+        default:
+            break
+        }
+        try await virtualMachine?.machine.stop()
+        await resetSlot()
     }
 
-    /// Handle a VM that was stopped from outside this object (for instance – by being shut down internally)
+    /// Resets the slot to a stopped status
     ///
     /// No need to do anything except some internal bookkeeping
-    ///
+    /// - Parameter error: Error supplied if the VM crashed
     @MainActor
-    func flush() async throws {
-        self.virtualMachine = nil
-        self.status = .empty
-    }
+    func resetSlot(withError error: Error? = nil) async {
+        if let virtualMachine {
+            try? await VMManager.removeVM(name: virtualMachine.config.handle)
```

Comment:
Could be worth to use `try` instead of `try?` and wrap it all in a `do { … } catch { }` to log the error about failing to delete the VM file, even if we don't want the overall method to `throw`.

Reply:
PS: If that pattern starts to become common through the code, one could consider a helper method on `Logger`, e.g.
```swift
Logger.helper.catchAndLogErrors {
  try await VMManager.removeVM(name: virtualMachine.config.handle)
}
```
With implementation that'd probably look something like this:
```swift
extension Logger {
  func catchAndLogErrors(block: () async throws -> Void) {
    do {
      try await block()
    } catch let error {
      log("\(error)", .error)
    }
  }
}
```

Reply:
Good idea, this could possibly even help with this mystery hang.